samples/authUserFlow.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+function main(projectId, clientId, clientSecret, refreshToken) {
+  // [START bigquery_auth_user_flow]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'PROJECT ID'
+  // const clientId = 'CLIENT ID'
+  // const clientSecret = 'CLIENT SECRET'
+  // const refreshToken = 'REFRESH TOKEN'
```

Comment:
I want to be a little more careful about refresh tokens in particular. Client ID & secret only identify an app, so it's okay to embed those in code, but refresh tokens identify a user so it's not appropriate to embed in code.

Is there a way we can encourage people to pass these in via command-line or environment variable?

Reply:
I assume we'll have some other sample code somewhere for how to get a refresh token?

In the Python example, we have folks paste in an authentication code to standard-in, which we then exchange for a refresh token.